src/metabase/sync.clj

```diff
@@ -34,6 +34,22 @@
      [:name       :string]
      [:steps      [:maybe [:sequential sync-util/StepNameWithMetadata]]]]]])
 
+(def ^:private phase->fn
+  {:metadata     sync-metadata/sync-db-metadata!
+   :analyze      analyze/analyze-db!
+   :field-values field-values/update-field-values!})
+
+(defn- scan-phases [scan]
+  (if (not= :full scan)
+    [:metadata]
+    [:metadata :analyze :field-values]))
+
+(defn- do-phase! [phase database]
+  (let [f      (phase->fn phase)
+        result (f database)]
+    (when-not (instance? Throwable result)
+      (assoc result :name (name phase)))))
```

Comment:
If we don't want to swallow errors here, this is where we can instead wrap and re-throw the exception.

Reply:
i think we want to swallow errors and continue. swallowing errors was how it used to work even before this `sync-operation` was first introduced [7 years ago](https://github.com/metabase/metabase/commit/206c1430e22e688953872bf34f7de704e2e80048), and I can't see a reason to change that now